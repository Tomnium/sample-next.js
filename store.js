import { createStore } from 'redux';
import { HYDRATE } from 'next-redux-wrapper';

const initialState = {
    posts:[],
}

export const reducer = (state = initialState,action) => {
    switch (action.type) {

        case HYDRATE:{
            return {
                ...action.payload
            }
        }
        case 'POSTS_SET_DATA':{
            return {
                ...state,
                posts:action.payload,
            }
        }

        default: {
            return state;
        }
    }
}



const store = createStore(reducer);

export default store;






