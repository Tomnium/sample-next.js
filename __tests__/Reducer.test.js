import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { render } from '@testing-library/react';
import { reducer } from './../store';
import Posts from './../pages/posts';
import '@testing-library/jest-dom/extend-expect';

const renderWithRedux = (
    component,
) => {
    const store = createStore(reducer)
    return {
        ...render(<Provider store={store}>
            {component}
        </Provider>),
        store,
    }
}


describe('Redux testing',() => {


    let component;
    beforeEach(() => component = renderWithRedux(<Posts/>))

    it("checks initial state",() => {
        const { store } = component
        expect(store.getState().posts.length).toEqual(0)
    })

    it("sets posts data and renders them",() => {
        const { store } = component
        const posts = [
            {
                id:'1',
                title:'Title',
                body:'Description',
                userId:1,
            },
            {
                id:'2',
                title:'Title',
                body:'Description',
                userId:2,
            }
        ]
        store.dispatch({ type:'POSTS_SET_DATA',payload:posts })
        expect(store.getState().posts.length).toEqual(2)
        expect(component.asFragment(component)).toMatchSnapshot()
    })
});


