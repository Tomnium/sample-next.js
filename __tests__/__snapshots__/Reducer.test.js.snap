// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux testing sets posts data and renders them 1`] = `
<DocumentFragment>
  <div
    class="undefined"
  >
    <div>
      <nav>
        <a
          href="/"
        >
          Main
        </a>
        <a
          href="/posts"
        >
          Posts
        </a>
        <a
          href="/about"
        >
          About
        </a>
      </nav>
    </div>
  </div>
  <main>
    <div>
      <h1
        class="sas"
      >
        Posts
      </h1>
      <div>
        <a
          href="/posts/1"
        >
          Title
        </a>
      </div>
      <div>
        <a
          href="/posts/2"
        >
          Title
        </a>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
